source.lua

```diff
@@ -1,10 +1,19 @@
 --@ module = true
 local repeatUtil = require('repeat-util')
 
-liquidSources = liquidSources or {}
+liquidSources = json.decode(persist.GlobalTable[GLOBAL_KEY] or {})
```

Comment:
you'll also need to define hooks to load the state when a map is loaded and re-register with repeat-util

Reply:
Apologies for creating this prematurely, it's been awhile since I've used GitHub and I was just refamiliarizing myself with it! I'll avoid creating `DRAFT/WIP` PRs in the future :sweat_smile: 

I'll be pushing more updates soon

Reply:
no worries! it's fine to create a draft PR. It's a good way to get continuous feedback when trying something new. There's a button for that, though -- you don't have to put DRAFT in the title. I'll click it for you.
![image](https://github.com/DFHack/scripts/assets/977482/1e72addb-0aaf-44ee-a93c-7a53757b629d)

Reply:
I appreciate the help!

So as things currently are, I have the Sources persisting between saves as long as the game isn't closed, but closing and relaunching the game causes the Sources to all be removed again.

Do I need to hook into something else? :thinking: 

Also I'm happy to hear any nitpicks and cleanup recommendations for my changes, this is my first endeavor into DFHack and I'm happy to learn how y'all do things.